src/figurl/useSyncHive.ts

```diff
@@ -0,0 +1,135 @@
+import { JSONStringifyDeterministic } from "kachery-js/types/kacheryTypes"
+import { useCallback, useMemo, useState } from "react"
```

Comment:
Just noting that the comments in this class make it super clear to follow what's going on. That's awesome!